packages/office-ui-fabric-react/src/components/DatePicker/DatePicker.tsx

```diff
@@ -290,14 +290,21 @@ export class DatePicker extends BaseComponent<IDatePickerProps, IDatePickerState
   private _onSelectDate(date: Date) {
     const { formatDate, onSelectDate } = this.props;
 
+    if (this.props.calendarProps && this.props.calendarProps.onSelectDate) {
```

Comment:
Out of curiosity, why do we have onSelectDate and calendarProps.onSelectDate?

Reply:
according to Drew, we are going to be passing calendarProps.onSelectDate so that we know whether the date was selected by the actual calendar drop down or whether it was typed, he needs to set a flag in the consumer. This was always an accepted parameter but before this change we were just overriding and ignoring the calendarProps.onSelectDate, no one used it so no one noticed before.